src/ros/Publisher.lua

```diff
@@ -55,7 +55,7 @@ function Publisher:delete()
     -- nothing to do. Connection object is not owned by this Publisher
 end
 
-function Publisher:publish(msg)
+function Publisher:publish(msg, topic_name)
```

Comment:
Isn't `topic_name` already passed to the ctor of this class?

It seems strange to me to create a `Publisher` publishing to a specific topic, only to then publish to a different topic by passing a different `topic_name` to `Publisher::publish(..)`?

Reply:
> Isn't topic_name already passed to the ctor of this class?

Oh yes, you are right

Yes, this makes much more sense. I've created vehicle-specific Publishers for `Odometry`, `LaserScan` and `Imu` messages.

Let me know if the changes are closer to what you expected. Thanks!

Reply:
This looks much cleaner :+1: